client/src/modules/stock/aggregated_consumption/modals/consumption_lot.modal.js

```diff
@@ -0,0 +1,212 @@
+
+  vm.entryType = Data.entry_type;
+  vm.isTransfer = (vm.entryType === 'transfer_reception');
+
+  // exposing method to the view
+  vm.submit = submit;
+  vm.cancel = cancel;
+
+  vm.onLotBlur = onLotBlur;
+  vm.onChanges = onChanges;
+  vm.onChangeQuantity = onChangeQuantity;
+  vm.onChangeUnitCost = onChangeUnitCost;
+  vm.onDateChange = onDateChange;
+
+  vm.isCostEditable = (vm.entryType !== 'transfer_reception');
+
+  const cols = [{
+    field : 'status',
+    width : 25,
+    displayName : '',
+    cellTemplate : 'modules/stock/entry/modals/templates/lot.status.tmpl.html',
+  }, {
+    field : 'start_date',
+    type : 'date',
+    cellFilter : `date:"${bhConstants.dates.format}"`,
+    width : 200,
+    displayName : 'STOCK.AGGREGATED_STOCK_CONSUMPTION.START_OF_CONSUMPTION',
+    headerCellFilter : 'translate',
+    cellTemplate : 'modules/stock/aggregated_consumption/templates/lot_movement_start_date.tmpl.html',
+  }, {
+    field : 'end_date',
+    type : 'date',
+    cellFilter : `date:"${bhConstants.dates.format}"`,
+    width : 200,
+    displayName : 'STOCK.AGGREGATED_STOCK_CONSUMPTION.END_OF_CONSUMPTION',
+    headerCellFilter : 'translate',
+    cellTemplate : 'modules/stock/aggregated_consumption/templates/lot_movement_date.tmpl.html',
+  }, {
+    field : 'quantity_consumed',
+    type : 'number',
+    displayName : 'STOCK.QUANTITY_CONSUMED',
+    headerCellFilter : 'translate',
+    aggregationType : uiGridConstants.aggregationTypes.sum,
+    aggregationHideLabel : true,
+    footerCellClass : 'text-right',
+    cellTemplate : 'modules/stock/aggregated_consumption/templates/consumed.tmpl.html',
+  }, {
+    field : 'quantity_lost',
+    type : 'number',
+    displayName : 'STOCK.QUANTITY_LOST',
+    headerCellFilter : 'translate',
+    aggregationType : uiGridConstants.aggregationTypes.sum,
+    aggregationHideLabel : true,
+    footerCellClass : 'text-right',
+    cellTemplate : 'modules/stock/aggregated_consumption/templates/lost.tmpl.html',
+  }, {
+    width : 75,
+    field : 'actions',
+    cellTemplate : 'modules/stock/entry/modals/templates/lot.actions.tmpl.html',
+  }];
+
+  vm.gridOptions = {
+    appScopeProvider : vm,
+    enableSorting : false,
+    enableColumnMenus : false,
+    showColumnFooter : true,
+    fastWatch : true,
+    flatEntityAccess : true,
+    data : vm.form.rows,
+    columnDefs : cols,
+    onRegisterApi,
+  };
+
+  function init() {
+    if (cache.enableFastInsert) {
+      vm.enableFastInsert = cache.enableFastInsert;
+    }
+
+    if (vm.form.rows.length) { return; }
+
+    vm.form.addItem();
+  }
+
+  function onRegisterApi(api) {
+    vm.gridApi = api;
+  }
+
+  /**
+   * @method onLotBlur
+   *
+   * @description
+   * if the fast insert option is enable do this :
+   * - add new row automatically on blur
+   * - set the focus in the new row
+   * @param {string} rowLot the row.entity.lot string
+   */
+  function onLotBlur(rowLot) {
+    if (vm.enableFastInsert && rowLot) {
```

Comment:
Do you ever use `enableFastInsert`?  I assume this code must be copied from somewhere.  We should probably remove anything that isn't critical for these modules.

This can be done after merge.